archaius2-core/src/main/java/com/netflix/archaius/DefaultAppConfig.java

```diff
@@ -253,8 +224,8 @@ public DefaultAppConfig(Builder builder) {
                     .withStrInterpolator(getStrInterpolator())
                     .build();
                 
-            super.addConfig(loader.newLoader().withName(APPLICATION_LAYER).load(builder.configName));
-            super.addConfig(library = new CascadingCompositeConfig(LIBRARY_LAYER));
+            root.addConfig(loader.newLoader().withName(APPLICATION_LAYER).load(builder.configName));
```

Comment:
I don't think this actually solves the problem Brent was asking about. He wants to set the Config used for the APPLICATION layer, or at least be able to add a Config to it if it was a composite.